### ISSUE TITLE
False positive isinstance-second-argument-not-valid-type when testing enum.Enum

### ISSUE THREAD

**Issue by jeromeag** [PyCQA/pylint#8327]
### Bug description

"""just to test pylint isinstance-second-argument-not-valid-type"""
from enum import Enum

Direction = Enum('Direction', ['LEFT', 'RIGHT', 'BOTH'])

def check_direction(direction):
    """just to test pylint isinstance-second-argument-not-valid-type"""
    if not isinstance(direction, Direction):
        raise TypeError(f'Not a RUKind ("{direction}" is a {type(direction)})')

### Command used

```shell
poetry run pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:8:11: W1116: Second argument of isinstance is not a type (isinstance-second-argument-not-valid-type)

------------------------------------------------------------------
Your code has been rated at 8.00/10 (previous run: 6.00/10, +2.00)
```

### Expected behavior

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 8.00/10, +2.00)

### Pylint version

```shell
pylint 2.16.2
astroid 2.14.2
Python 3.9.10 (main, Jan 20 2022, 21:37:52) 
[GCC 11.2.0]
```

### OS / Environment

Cygwin 
Poetry (version 1.3.2)

**Comment by nickdrozd** [comment]
The `isinstance-second-argument-not-valid-type` cannot handle enums correctly at all. Besides the false positive reported here, it also exhibits false negatives. Consider this code:

```python
from enum import Enum

Direction = Enum('Direction', ['Left', 'Right', 'Both'])

# False positives
print(isinstance(0, Direction))
print(isinstance(Direction.Left, Direction))

# False negatives
print(isinstance(0, Direction.Left))
print(isinstance(Direction.Left, Direction.Left))
```

The first two `isinstance` calls are flagged, but they shouldn't be, since they run fine. The second pair of calls are not flagged, but they should be, since they fail at runtime.
